arch/xtensa/soc/sample_controller/linker.ld

```diff
@@ -48,7 +49,8 @@ MEMORY
   iram0_19_seg :                      	org = 0x40000400, len = 0x1FC00
   srom0_seg :                         	org = 0x50000000, len = 0x300
   srom1_seg :                         	org = 0x50000300, len = 0xFFFD00
-  sram0_seg :                         	org = 0x60000000, len = 0x4000000
+  sram0_seg :                         	org = CONFIG_SRAM_BASE_ADDRESS,
+					len = CONFIG_SRAM_SIZE*1K
```

Comment:
Said this earlier, will repeat: I know this sounds like it's a cool feature to unify memory mapping across all the targets.  It's not.  It's a footgun.  We **will** have bugs due to this decoupling, where systems inexplicably fail due to their memory map being "configured" when it never should have been configurable in the first place.

This is the linker script.  It is the final (and should be the only) source for authority when it comes to device memory layout.  Putting it anywhere else is only going to hurt us.

I'm not going to -1 an in-progress architectural change over this.  But I won't shut up that I think it's a terrible idea.